src/sources/apache_metrics/parser.rs

```diff
@@ -0,0 +1,852 @@
+use crate::event::metric::{Metric, MetricKind, MetricValue};
+use chrono::{DateTime, Utc};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt;
+
+/// Parses the text output from Apache's mod_status and returns:
+///
+/// - A list of metrics generated from the output
+/// - A list of parse errors that were encountered
+///
+/// # Arguments
+///
+/// - `payload` - the mod_status output
+/// - `namespace` - the namespace to put the generated metrics in
+/// - `now` - the time the payload was fetched
+/// - `tags` - any base tags to apply to the metrics
+pub fn parse(
+    payload: &str,
+    namespace: &str,
+    now: DateTime<Utc>,
+    tags: Option<&BTreeMap<String, String>>,
+) -> (Vec<Metric>, Vec<ParseError>) {
```

Comment:
I'm curious to get feedback or thoughts on how to model this better where the parse function wants to return metrics _and_ errors rather than simply one or the other. This was a design choice I made to still publish metrics that we could parse even if there was an error with one of them.

Reply:
I wonder if an iterator-based approach would work well?

The iterator would return `Option<Result<Metric, ParseError>>`. The caller can continue iterating, even if an error is returned, until no more metrics are available.

You can't have it return a metric _and_ error simultaneously, but I don't believe that's needed?

Regardless, what you have here works nicely, so I wouldn't worry too much about changing this.

Reply:
Hmm, yeah, I think I like that better. I'll give it a shot.